app/models/provisioning_template.rb

```diff
@@ -105,42 +105,51 @@ def self.find_template(opts = {})
     template.is_a?(ProvisioningTemplate) ? template : nil
   end
 
-  def self.build_pxe_default(renderer)
-    return [422, _("No TFTP proxies defined, can't continue")] if (proxies = SmartProxy.with_features("TFTP")).empty?
+  def self.find_global_default_template(name, kind)
+    ProvisioningTemplate.joins(:template_kind).where(:name => name, "template_kinds.name" => kind).first
+  end
 
+  def self.build_pxe_default(renderer)
+    return [422, _("No TFTP proxies defined, can't continue"), []] if (proxies = SmartProxy.with_features("TFTP")).empty?
     error_msgs = []
+    info_msgs = []
     TemplateKind::PXE.each do |kind|
-      default_template_name = "#{kind} global default"
-      if (default_template = ProvisioningTemplate.find_by_name(default_template_name)).nil?
-        error_msg = _("Could not find a Configuration Template with the name \"%s global default\", please create one.") % kind
+      setting_name = "global_#{kind}"
```

Comment:
I'd suggest extracting this to a dedicated method to reduce complexity of `build_pxe_default`.